### ISSUE TITLE
flang --version returns "clang" instead of "flang"

### ISSUE THREAD

**Issue by rouson** [flang-compiler/flang#332]
Invoking `flang ` with the `--version` argument identifies the compiler as `clang`.  Is this the intended behavior?

```bash
$flang --version
clang version 5.0.0 (tags/RELEASE_500/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.3.1/llvm-5.0.0-x55pnrm3tdrrmnpoxcxxzcgb46q3ee6q/bin

```

**Comment by junghans** [comment]
`flang` is part of the `clang` package, so that makes kind of sense!

**Comment by rouson** [comment]
@junghans Thanks for the feedback.   The [clang web site] states, "The goal of the Clang project is to create a new C based language front-end: C, C++, Objective C/C++, OpenCL C and others for the LLVM compiler."  Possibly Fortran falls under "others," but I don't see any mention of Fortran or flang anywhere on the clang web site.  Of course, flang and clang are related in that they both target LLVM as the back-end and I do see that there's a [clang repository] under the [flang-compiler] organizational account so maybe the reverse is true: clang is part of flang-compiler, but I don't understand how flang is part of clang.   

FYI, part of the reason I'm asking is for purposes of detecting the compiler identity and version inside a script.  In that context, it's useful, though not essential, for `flang --version` to return something that includes the string flang or at least something other than `clang`.

[clang web site]: https://clang.llvm.org
[clang repository]: https://github.com/flang-compiler/clang 
[flang-compiler]: https://github.com/flang-compiler

**Comment by junghans** [comment]
Can you use `flang1` for that?

**Comment by rouson** [comment]
Please explain.  I see that `flang1` is in the same installation path as `flang`, but I don't know how to use `flang1`.  Invoking it with no arguments appears to hang.  Invoking it with `--help` or `--version` seg faults.  Invoking it on a Fortran program generates something that resembles assembly.

**Comment by xoviat** [comment]
`flang1` is not intended to be invoked by the end-user. It should only be invoked by the developers of this project (which includes contributors!) or other executables.

TL;DR this is a bug, but there are *much* more important bugs to fix.

**Comment by ponnamsairam** [comment]
Hey @rouson  
As the intention of developing FLANG front-end is for using LLVM-CLANG compiler as backend to provide support for running FORTRAN programs so here basically FLANG supports all CLANG compiler options and specifically there are some more compiler options for FLANG you can check them by $flang -help
